internal/backend/fs.go

```diff
@@ -310,21 +311,34 @@ func (s *storageFS) DeleteObject(bucketName, objectName string) error {
 	return os.Remove(path)
 }
 
-// PatchObject patches the given object metadata.
-func (s *storageFS) PatchObject(bucketName, objectName string, metadata map[string]string) (StreamingObject, error) {
+// func (s *storageFS) PatchObject(bucketName, objectName string, metadata map[string]string) (StreamingObject, error) {
+func (s *storageFS) PatchObject(bucketName, objectName string, attrsToUpdate ObjectAttrs) (StreamingObject, error) {
 	obj, err := s.GetObject(bucketName, objectName)
 	if err != nil {
 		return StreamingObject{}, err
 	}
 	defer obj.Close()
-	if obj.Metadata == nil {
-		obj.Metadata = map[string]string{}
-	}
-	for k, v := range metadata {
-		obj.Metadata[k] = v
+
+	currObjValues := reflect.ValueOf(&(obj.ObjectAttrs)).Elem()
+	currObjType := currObjValues.Type()
+	newObjValues := reflect.ValueOf(attrsToUpdate)
+	for i := 0; i < newObjValues.NumField(); i++ {
+		if reflect.Value.IsZero(newObjValues.Field(i)) {
+			continue
+		} else if currObjType.Field(i).Name == "Metadata" {
+			if obj.Metadata == nil {
+				obj.Metadata = map[string]string{}
+			}
+			for k, v := range attrsToUpdate.Metadata {
+				obj.Metadata[k] = v
+			}
+		} else {
+			currObjValues.Field(i).Set(newObjValues.Field(i))
+		}
```

Comment:
nit: could we abstract this in a helper function? It's the same in both backends, right?